jbmc/src/java_bytecode/load_method_by_regex.cpp

```diff
@@ -0,0 +1,70 @@
+
+/*******************************************************************\
+
+Module: Java Bytecode
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include "load_method_by_regex.h"
+
+#include <regex>
+
+#include <util/symbol_table.h>
+
+/// Create a load_method_by_regext matching the pattern. If the pattern doesn't
+/// include a colon for matching the descriptor, append a `:\(.*\).*` to the
+/// regex. Note this will mean all overloaded methods will be marked as extra
+/// entry points for CI lazy loading.
+/// If the pattern doesn't include the java:: prefix, prefix that
+/// \param pattern: The user provided pattern
+load_method_by_regext::load_method_by_regext(const std::string &pattern)
+{
+  std::string modified_pattern = pattern;
+  if(does_pattern_miss_descriptor(pattern))
+    modified_pattern += R"(:\(.*\).*)";
+
+  if(!has_prefix(pattern, "java::"))
+    modified_pattern = "java::" + modified_pattern;
+
+  regex = std::regex{modified_pattern};
+}
+
+/// Get the methods that match the regex
+/// \param symbol_table: The symbol table to look through
+/// \return: The method identifiers that should be loaded
+std::vector<irep_idt>
+load_method_by_regext::extra_methods(const symbol_tablet &symbol_table) const
+{
+  std::vector<irep_idt> matched_methods;
+  for(const auto &symbol : symbol_table.symbols)
+  {
+    if(symbol.second.is_function())
+    {
+      const std::string symbol_name = id2string(symbol.first);
+      if(std::regex_match(symbol_name, regex))
```

Comment:
Combine conditions, flatten the loop? Also this is another single-method-class that could've been replaced by a lambda.

Reply:
Think is is clearer as two separate checks.